exec/java-exec/src/main/java/org/apache/drill/exec/planner/fragment/DistributionAffinity.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.planner.fragment;
+
+/**
+ * Describes an operator's endpoint assignment requirements.
+ */
+public enum DistributionAffinity {
+  /**
+   * No affinity to any endpoints. Operator can run on any endpoint.
+   */
+  NONE(0, SoftAffinityFragmentParallelizer.INSTANCE),
+
+  /**
+   * Operator has soft distribution affinity to one or more endpoints. Operator performs better when fragments are
+   * assigned to the endpoints with affinity, but not a mandatory requirement.
+   */
+  SOFT(1, SoftAffinityFragmentParallelizer.INSTANCE),
+
+  /**
+   * Hard distribution affinity to one or more endpoints. Fragments having the operator must be scheduled on the nodes
+   * with affinity.
+   */
+  HARD(2, HardAffinityFragmentParallelizer.INSTANCE);
+
+  private int level;
+  private FragmentParallelizer fragmentParallelizer;
+
+  DistributionAffinity(final int level, final FragmentParallelizer fragmentParallelizer) {
+    this.level = level;
+    this.fragmentParallelizer = fragmentParallelizer;
+  }
+
+  public FragmentParallelizer getFragmentParallelizer() {
+    return fragmentParallelizer;
+  }
+
+  /**
+   * Is the current DistributionAffinity less or equal restrictive than the given DistributionAffinity?
+   * @param distributionAffinity
+   * @return
+   */
+  public boolean isLessOrEqualRestrictive(final DistributionAffinity distributionAffinity) {
```

Comment:
name suggestion change: lessThanOrEqualTo? At the same time, this enum implements the Comparable<DistributionAffinity> interface, so as long as the level is the same as the index, it might be enough...